### PR TITLE
VIH-7803 staffmember in judge joh and participant waiting room

### PR DESCRIPTION
**Pull request by PrasannaDommalapati**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-7803

### Change description ###

- Displaying staff members in the judge, Joh and participant waiting rooms

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.html [hunk 1]
```diff
@@ -1,154 +1,175 @@
...
-        <div id="p-row-{{ participant.id }}-panel-member"
-          class="govuk-grid-column-full">
-          <p id="p-{{ participant.id }}-name-panel-member"
-            class="govuk-heading-s govuk-!-margin-bottom-0">{{ participant.name }}</p>
-          <div class="vh-line"
-            *ngIf="!lastItem"></div>
+    <div *ngIf='staffMembers.length > 0'>
+        <div class='govuk-grid-row'>
+            <div class='govuk-grid-column-full'>
+                <h2 class='govuk-heading-s'>{{'individual-participant-status-list.staff-members' | translate}} </h2>
```
Comment (ZJianWang) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.html:65]: not sure if  need show numbers of the staff members for now like other type particilant

Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/individual-participant-status-list/individual-participant-status-list.component.html:65]: screenshot added ny neil have no number

**Review by ZJianWang — APPROVED** [review]
LGTM

**Review by robertbennett1998 — CHANGES_REQUESTED** [review]
Small changes but overall LGTM

File: VideoWeb/VideoWeb.AcceptanceTests/Features/StaffMember.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Feature: StaffMember
+	As a: Judge, Judicial Office Holder (panel member/winger), Participant
+  I want to: See the the Staff Member in the Waiting Room hearing list
+  So that: I can know who is in the hearing and what their role is
+
+Background:
+	#Given I have a hearing in 20 minutes time with staffmember
```
[the remark this answers is not in the document]
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb.AcceptanceTests/Features/StaffMember.feature:7]: forgot this completely :(

File: VideoWeb/VideoWeb.AcceptanceTests/Steps/WaitingRoomSteps.cs [hunk 3]
```diff
@@ -240,6 +239,12 @@ public void ThenTheUserCanSeeOtherParticipantsStatus()
             }
         }
 
+        [Then(@"the (participant|joh|judge) can see the list of staff members")]
+        public void ThenTheHaeringParticipantCanSeeListOfStaffMembers(string _)
```
[the remark this answers is not in the document]
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb.AcceptanceTests/Steps/WaitingRoomSteps.cs:243]: resolved

File: VideoWeb/VideoWeb.AcceptanceTests/Hooks/RemoveDataHooks.cs [hunk 4]
```diff
@@ -49,7 +49,12 @@ private static void DeleteTheHearing(TestApiManager api, Guid hearingId)
         private void ClearClosedConferencesForUser(TestApiManager api)
         {
             var response = api.GetConferencesForTodayJudge(_username);
-            var todaysConferences = RequestHelper.Deserialise<List<ConferenceForHostResponse>>(response.Content);
+            string jsonstr = response.Content;
```
[the remark this answers is not in the document]
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb.AcceptanceTests/Hooks/RemoveDataHooks.cs]: addressed by extracting to a private method

**Review by robertbennett1998 — DISMISSED** [review]
LGTM

**Review by ZJianWang — DISMISSED** [review]
looks good

File: VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs [hunk 5]
```diff
@@ -110,16 +110,16 @@ private static void AddPolicies(AuthorizationOptions options, IList<IProviderSch
...
-                AppRoles.VhOfficerRole, AppRoles.JudicialOfficeHolderRole, AppRoles.StaffMemberRole
+                AppRoles.VhOfficerRole, AppRoles.JudicialOfficeHolderRole, AppRoles.StaffMember
             };
 
             var rolePolicies = new Dictionary<string, string[]>
             {
                 [AppRoles.JudgeRole] = new[] { AppRoles.JudgeRole },
                 [AppRoles.VhOfficerRole] = new[] { AppRoles.VhOfficerRole },
-                ["Judicial"] = new[] { AppRoles.JudgeRole, AppRoles.JudicialOfficeHolderRole, AppRoles.StaffMemberRole },
+                ["Judicial"] = new[] { AppRoles.JudgeRole, AppRoles.JudicialOfficeHolderRole, AppRoles.StaffMember },
```
Comment (omio-ma) [inline VideoWeb/VideoWeb/Extensions/ConfigureAuthSchemeExtensions.cs:120]: Although this has already been like this, I think to keep consistent we should declare this as consts similarly to the other AppRoles

**Review by robertbennett1998 — DISMISSED** [review]
LGTM

**Review by robertbennett1998 — APPROVED** [review]
LGTM

File: VideoWeb/VideoWeb/Mappings/ClaimsPrincipalToUserProfileResponseMapper.cs [hunk 6]
```diff
@@ -69,6 +69,11 @@ private Role DetermineRoleFromClaims(ClaimsPrincipal user)
                 return Role.CaseAdmin;
             }
 
+            if (user.IsInRole(AppRoles.StaffMember))
```
[the remark this answers is not in the document]
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb/Mappings/ClaimsPrincipalToUserProfileResponseMapper.cs:67]: Removed

**Review by omio-ma — APPROVED** [review]

**Pull request merged**